inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_non_max_suppression.cpp

```diff
@@ -11,38 +11,30 @@
-                     target->get_friendly_name(), target->get_type_info(), ngraph::vpu::op::DynamicShapeResolver::type_info, 1);
-
-    const auto scores_shape = dsr1->input(1).get_source_output();
-
-    const auto index_num_classes = ngraph::opset3::Constant::create(ngraph::element::i64, {1}, std::vector<int64_t>{1});
-    const auto axis_num_classes = ngraph::opset3::Constant::create(ngraph::element::i64, {}, std::vector<int64_t>{0});
-    const auto num_classes = std::make_shared<ngraph::opset3::Gather>(scores_shape, index_num_classes, axis_num_classes);
-
-    const auto index_num_boxes = ngraph::opset3::Constant::create(ngraph::element::i64, {1}, std::vector<int64_t>{2});
-    const auto axis_num_boxes = ngraph::opset3::Constant::create(ngraph::element::i64, {}, std::vector<int64_t>{0});
-    const auto num_boxes = std::make_shared<ngraph::opset3::Gather>(scores_shape, index_num_boxes, axis_num_boxes);
-
-    VPU_THROW_UNLESS(target->inputs().size() > 2,  "DynamicToStaticShape transformation for {} expects at least 3 inputs", target);
-    // originally 3rd input is a scalar of any integer type, so we cast and unsqueeze it to 1D
-    const auto max_output_boxes_per_class = std::make_shared<ngraph::opset3::Convert>(std::make_shared<ngraph::opset3::Unsqueeze>(
-            target->input_value(2).get_node_shared_ptr(), ngraph::opset3::Constant::create(ngraph::element::i32, {1}, {0})), scores_shape.get_element_type());
-
-    const auto max_output_boxes_overall = std::make_shared<ngraph::opset3::Multiply>(max_output_boxes_per_class, num_classes);
-    const auto num_selected_boxes = std::make_shared<ngraph::opset3::Minimum>(num_boxes, max_output_boxes_overall);
-
-    const auto triplet_const = ngraph::opset3::Constant::create(scores_shape.get_element_type(), {1}, std::vector<int64_t>{3});
-    const auto output_shape = std::make_shared<ngraph::opset3::Concat>(ngraph::OutputVector{num_selected_boxes, triplet_const}, 0);
-
-    const auto copied = target->clone_with_new_inputs(target->input_values());
-    ngraph::replace_node(target, std::make_shared<ngraph::vpu::op::DynamicShapeResolver>(copied, output_shape));
+void dynamicToStaticNonMaxSuppression(std::shared_ptr<ngraph::Node> node) {
+    auto nms_4 = std::dynamic_pointer_cast<ngraph::op::v4::NonMaxSuppression>(node);
+    VPU_THROW_UNLESS(nms_4, "dynamicToStaticNonMaxSuppression transformation for {} of type {} expects {} as node for replacement",
+                     node->get_friendly_name(), node->get_type_info(), ngraph::op::v4::NonMaxSuppression::type_info);
+
+    auto staticShapeNMS = std::make_shared<ngraph::vpu::op::StaticShapeNonMaxSuppression>(
+            nms_4->input_value(0),
+            nms_4->input_value(1),
+            nms_4->input_value(2),
+            nms_4->input_value(3),
+            nms_4->input_value(4),
+            static_cast<const ngraph::vpu::op::StaticShapeNonMaxSuppression::BoxEncodingType>(nms_4->get_box_encoding()),
```

Comment:
It's unclear from changes in PR to me - why do you need `static_cast` here? Which type `get_box_encoding` returns?

Reply:
It is not needed anymore, thanks for catching. Resolved